### PR TITLE
[BACK-2019] Update data model to support uploading Tidepool Loop settings

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/BACK-2019
- Update pumpSettings with new fields
- Update cgmSettings with new fields
- Add controllerSettings
- Add deviceEvent/pumpSettingsOverride
- Remove applicationSettings
- Tests

### CODE REVIEW

**Review by toddkazakov — CHANGES_REQUESTED** [review]
Overall looks good. I think you should just rename the functions which generate random data instead of deprecating them, because this only adds tech debt.

File: data/types/settings/cgm/cgm.go [hunk 1]
```diff
@@ -117,6 +134,12 @@ func (c *CGM) Normalize(normalizer data.Normalizer) {
 
 	units := c.Units
 
+	if normalizer.Origin() == structure.OriginExternal {
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/types/settings/cgm/cgm.go:137]: Yes, I see. I will combine.

[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/types/settings/cgm/cgm.go:137]: Done.

**Comment by darinkrauss** [comment]
@toddkazakov  I removed the deprecated New* test functions that I changed.

File: data/blood/glucose/test/target.go [hunk 2]
```diff
@@ -10,7 +10,19 @@ import (
 	"github.com/tidepool-org/platform/test"
 )
 
+// DEPRECATED: Use RandomTarget
```
Comment (toddkazakov) [inline data/blood/glucose/test/target.go:13]: This should be renamed instead of deprecated

File: data/types/device/test/device.go [hunk 3]
```diff
@@ -3,12 +3,30 @@ package test
 import (
 	"github.com/tidepool-org/platform/data/types/device"
 	dataTypesTest "github.com/tidepool-org/platform/data/types/test"
+	"github.com/tidepool-org/platform/test"
 )
 
+// DEPRECATED: Use RandomDevice
```
Comment (toddkazakov) [inline data/types/device/test/device.go:9]: This should be renamed instead of deprecated

File: data/types/test/base.go [hunk 4]
```diff
@@ -15,32 +15,45 @@ import (
 	userTest "github.com/tidepool-org/platform/user/test"
 )
 
+// DEPRECATED: Use RandomBase
```
Comment (toddkazakov) [inline data/types/test/base.go:18]: This should be renamed instead of deprecated

**Review by toddkazakov — CHANGES_REQUESTED** [review]
There are few more deprecated functions that should be renamed instead of deprecated

**Comment by darinkrauss** [comment]
@toddkazakov Sorry, my bad. I made the changes to a different branch and pushed that. Will move to this branch and re-request review.

**Comment by darinkrauss** [comment]
@toddkazakov Okay, commits added here. Should be good now.

**Review by toddkazakov — APPROVED** [review]
LGTM 👍

**Pull request merged**